### PR TITLE
MAINT,DOC: Readability improvements and cleanup for f2py

### LINKED ISSUE #19910: f2py dead code elimination
## Documentation

Not 100% a documentation issue per-se, but currently `f2py`:
- outputs links which lead to nowhere
- includes code (tests) which are not part of the `numpy` system
  - Legacy holdovers from when `f2py` was merged in

These should be cleaned up.

### PR DESCRIPTION
**Pull request by HaoZeke**
OK. This is ready now. It is essentially a very minor PR to make the code seem less out of date in production by:
- Updating dead URLs to point to valid Archive links
- Cleaning up some docstrings
- Removing tests which don't run
- Removing other parts of the code which are not relevant to `numpy`
Let me know if this is too aggressive.

Closes #19910.

### CODE REVIEW

File: tools/linter.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 EXCLUDE = (
     "numpy/typing/tests/data/",
     "numpy/__config__.py",
+    "numpy/f2py",
```
Comment (HaoZeke) [inline tools/linter.py:18]: RE: This seems rather harsh, but the `f2py` code has many lines which fail the linter; and this is the cleanest way to deal with this for now.

**Comment by HaoZeke** [comment]
RE: Removing tests

^ Another seemingly potentially aggressive cleaning step, this is a good thing to have before we move forward with the new test suite.

File: numpy/f2py/setup.py [hunk 2]
```diff
@@ -39,8 +39,8 @@ def configuration(parent_package='', top_path=None):
     config = configuration(top_path='')
     config = config.todict()
 
-    config['download_url'] = "http://cens.ioc.ee/projects/f2py2e/2.x"\
-                             "/F2PY-2-latest.tar.gz"
+    config['download_url'] = "https://web.archive.org/web/20140822061353/"\
+        "http://cens.ioc.ee/projects/f2py2e/2.x/F2PY-2-latest.tar.gz"
```
Comment (HaoZeke) [inline numpy/f2py/setup.py:43]: Not sure we need to provide this; am ambivalent about removal though.

**Comment by charris** [comment]
> RE: Removing tests

Are they currently unused? Are you putting together another test set?

File: numpy/f2py/setup.py (hunk 2 above)
Comment (charris) [reply numpy/f2py/setup.py:43]: I think it is outdated. @pearu comment?

File: tools/linter.py (hunk 1 above)
Comment (charris) [reply tools/linter.py:18]: Seems fair. I wish there was a python formatter that did what we want.

Comment (melissawm) [reply tools/linter.py:18]: We did talk about running a linter through all the code at some point. Maybe we can do that in a subsequent PR

File: numpy/f2py/setup.py (hunk 2 above)
Comment (charris) [reply numpy/f2py/setup.py:43]: Thinking about it more, I'd just remove it. F2PY is now distributed with NumPy and has undergone significant changes.

**Comment by HaoZeke** [comment]
> > RE: Removing tests
> 
> Are they currently unused? Are you putting together another test set?

This set is unused. There are `tests` under `numpy/f2py/tests` which are run with `runtests.py`. There are plans to update these for consistency; basically to extract the `fortran` snippets into their own file instead of keeping them in the `python` code.

Here is an extract of the logic behind this:

 At the moment it seems like the newer tests use code  and suffix to put the fortran source inside the `test_*.py` (https://github.com/numpy/numpy/commit/8336e76a8655494ad9ac641651fb411d34fa844b#diff-beaa766b50d9c2a8df83b293fa468e786d9ffb8b10d3f2a44d46d81084abd022)

However, the older tests had (IMO) a nicer structure with the source code in a separate f/90 file (even though the older test style can be cleaned up now).

As an example of the older test approach it is the one in test_mixed.py or test_common.py (https://github.com/numpy/numpy/commit/8042526ba48c6d817d32300c1ecbea95d395c62e).
For new tests, I'm thinking the `split .f90` and `test_*.py`  style should be used (with `pathlib` instead of the `_path helper`)

**Comment by charris** [comment]
In case it is useful for writing tests, Fortran Lang uses [fypp](https://fypp.readthedocs.io/en/stable/fypp.html) templates. I suspect you know that already :)

**Comment by charris** [comment]
Thanks @HaoZeke .

**Pull request merged**